test/examples/political_democracy/constructor.jl

```
@@ -1,10 +1,12 @@
 using Statistics: cov, mean
-using Random
+using Random, NLopt
 
 ############################################################################################
 ### models w.o. meanstructure
 ############################################################################################
 
+semoptimizer = SemOptimizer(engine = opt_engine)
+
 model_ml = Sem(specification = spec, data = dat, optimizer = semoptimizer)
```

I'm not sure this works - since the optimizer now lives outside of the Sem object, the keyword `optimizer` will just be ignored, and the optimizer has to be added to each call of sem_fit instead, right?

---

No, it doesn't work -- I forgot to update it. Should be fixed now.
That, sort of, highlights one limitation of the current API: it's based on passing arguments via keywords through the hierarchy of ctor calls.
And non-matching keywords are silently ignored, because each ctor cannot know whether it's an incorrect keyword or it is the correct keyword for another ctor.
So the bug like this was not automatically detected by the tests.

---

That is probably the topic for a separate issue, but I think the main ctors for `SemSpecification`, `SemObserved` and `Sem` that do all of the actual construction work should get the essential inputs as positional arguments (that is also important for type-dependent dispatch) and don't do keyword arguments passthrough.
Then it might be possible to have a simple single ctor-like function for `Sem()` that internally creates all 3 objects like it does now, but with more constrained set of keywords.

Also, explicitly creating `SemSpecification`, `SemObserved` and explicitly passing them to `Sem` might in the end be not so much more complex for the user.
The real design problem with that approach is that `SemSpec` and `SemObserved` are independent objects, but they are constrained to have the same set and same order of observed variables.
Maybe it is possible to utilize `ScopedValues` from 1.11, so that the user explicitly calls `Sem(SemSpec(...; kwargs1...), SemObserved(...; kwargs2); kwargs3...)`, but the observed variables order is implicitly passed to `SemSpec()` and `SemObserved()` via `ScopedValues`.

---

I references this in a issue for now, so we can revisit it at a later point 👍🏽